### PR TITLE
fix: reimplement crash logs written to file on disk

### PR DESCRIPTION
**Pull request by armcknight**
This is helpful to debug crash handling, which can't be done with the debugger attached. Reading log files is easier than getting the same info out of the system console.

I removed the file-based local level definitions, so that it's completely centralized. 

Also moved the macro def that effectively switches between console and file logging to the header so that it can be used to conditionally compile the other parts that set up file-based logging. Those pieces of code have been brought back after being removed in #2440 

#skip-changelog

### CODE REVIEW

**Review by philipphofmann — COMMENTED** [review]
Looking good so far.

**Comment by armcknight** [comment]
I'm not sure why I opened this as a draft, it should be ready for review. I'm not sure any of the test failures are related.

File: Sources/SentryCrash/Recording/Tools/SentryCrashLogger.h [hunk 1]
```diff
@@ -129,115 +129,139 @@
...
+#define HDR_SentryCrashLogger_h
 
-#    ifdef __cplusplus
+#ifdef __cplusplus
 extern "C" {
-#    endif
+#endif
 
-#    include <stdbool.h>
+#include <stdbool.h>
```
Comment (brustolin) [inline Sources/SentryCrash/Recording/Tools/SentryCrashLogger.h:170]: m: We had some problems before by declaring headers after `extern "C"`. Do you think we can move this up?

**Review by brustolin — APPROVED** [review]
Looks good!

**Review by philipphofmann — APPROVED** [review]
Again, LGTM.

**Pull request merged**